client/my-sites/checkout/checkout-thank-you/hundred-year-thank-you/index.tsx

```diff
@@ -203,23 +223,27 @@ export default function HundredYearThankYou( {
 					'The %(planTitle)s for %(domain)s is active. Our Premier Support team will be in touch by email shortly to schedule a welcome session and walk you through your exclusive benefits. We’re looking forward to supporting you every step of the way.',
 					{
 						args: {
-							domain: registeredDomain?.domain || siteSlug,
+							domain: targetDomain?.domain || siteSlug,
 							planTitle: getPlan( PLAN_100_YEARS )?.getTitle() || '',
 						},
 					}
 			  )
 			: translate(
-					'Your 100-Year Domain %(domain)s has been registered. Our Premier Support team will be in touch by email shortly to schedule a welcome session and walk you through your exclusive benefits. We’re looking forward to supporting you every step of the way.',
+					'Your 100-Year Domain %(domain)s has been %(action)s. Our Premier Support team will be in touch by email shortly to schedule a welcome session and walk you through your exclusive benefits. We’re looking forward to supporting you every step of the way.',
 					{
 						args: {
-							domain: registeredDomain?.domain || siteSlug,
+							domain: targetDomain?.domain || siteSlug,
+							action:
+								productSlug === domainProductSlugs.DOTCOM_DOMAIN_REGISTRATION
+									? translate( 'registered' )
+									: translate( 'transferred' ),
 						},
 					}
 			  );
 
 	return (
 		<>
-			{ siteId && <QuerySiteDomains siteId={ siteId } /> }
+			{ siteId && ! siteDomains.length && <QuerySiteDomains siteId={ siteId } /> }
```

Comment:
Is this additional check necessary?

Reply:
Yup - we only want to query the site domains if we don't have them. 
Otherwise, we'll end up querying it every time, even if it's available already. It's a small optimization but worth it in my opinion.

Reply:
Ah I see, I wondered if there was already logic in the `QuerySiteDomians` component to prevent that